gen/calc.py

```diff
@@ -261,7 +261,14 @@ def validate_dcos_overlay_network(dcos_overlay_network):
 
 
 def calculate_oauth_available(oauth_enabled):
-    return oauth_enabled
+    if oauth_enabled in ["[[[variables('oauthEnabled')]]]", '{ "Ref" : "OAuthEnabled" }']:
```

Comment:
We're duplicating these values outside their context (AWS, Azure). Would it make sense to define them in their respective modules, import them into `gen.calc`, and refer to them by name?